backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportHeaderGenerator.kt

```diff
@@ -1054,19 +1046,36 @@ abstract class ObjCExportHeaderGenerator internal constructor(
         return ObjCExportedInterface(generatedClasses, extensions, topLevel, headerLines, namer, mapper)
     }
 
+    fun getExportStubs(): ObjCExportedStubs =
+        ObjCExportedStubs(classForwardDeclarations, protocolForwardDeclarations, stubs)
+
     protected abstract fun reportWarning(text: String)
 
     protected abstract fun reportWarning(method: FunctionDescriptor, text: String)
 
     protected open fun getAdditionalImports(): List<String> = emptyList()
 
-
-    fun translateModule(): List<Stub<*>> {
+    fun translateModule() {
         // TODO: make the translation order stable
         // to stabilize name mangling.
+        translateBaseDeclarations()
+        translatePackageFragments()
+        translateExtraClasses()
+    }
 
-        stubs += translator.generateBaseDeclarations()
+    fun translateExtraClasses() {
+        while (extraClassesToTranslate.isNotEmpty()) {
+            val descriptor = extraClassesToTranslate.iterator().run { next().also { remove() } }
```

Comment:
Is using iterator here an optimization?

Reply:
Yes, just something small I noticed. The previous version required the calculation of the descriptors hashcode, whereas this doesn't.
If you prefer the previous version, please advise.

Reply:
> If you prefer the previous version, please advise.

Absolutely, I do prefer!
AFAIK, descriptor's `hashCode` is identity hash code, so it shouldn't cost much. On the other hand, this new variant is barely readable to me.

Reply:
👌